`S3Exception: null` when using the `s3.Downloads` task

### Expected Behavior

reproducer:

```yaml
id: downloadObjects
namespace: dev

variables:
  bucket: kestraio

tasks:
  - id: download
    type: io.kestra.plugin.aws.s3.Downloads
    bucket: "{{vars.bucket}}"
    prefix: inbox/
    filter: FILES
    action: MOVE
    moveTo: 
      bucket: "{{vars.bucket}}"
      key: archive/

  - id: printObjects
    type: io.kestra.core.tasks.log.Log
    message: "found objects {{outputs.download.objects}}"

taskDefaults:
  - type: io.kestra.plugin.aws.s3.Downloads
    values:
      accessKeyId: "{{ secret('AWS_ACCESS_KEY_ID') }}"
      secretKeyId: "{{ secret('AWS_SECRET_ACCESS_KEY') }}"
      region: "{{ secret('AWS_DEFAULT_REGION') }}"
```



### Actual Behaviour

error message:

```bash
Found '3' keys on kestraio with regexp='null', prefix=inbox/
software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 400, Request ID: 7M5CBX6N4PHA05PS, Extended Request ID: 2KQkAT2rX5aX4za54NbzkIbhGQaPHJjmIBzNrKeR9Zc8zeoJGuM+4JlUf2Y4sg6e2JmrEUFB2yA=)
null (Service: S3, Status Code: 400, Request ID: 7M5CBX6N4PHA05PS, Extended Request ID: 2KQkAT2rX5aX4za54NbzkIbhGQaPHJjmIBzNrKeR9Zc8zeoJGuM+4JlUf2Y4sg6e2JmrEUFB2yA=)
java.util.concurrent.ExecutionException: software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 400, Request ID: 7M5CBX6N4PHA05PS, Extended Request ID: 2KQkAT2rX5aX4za54NbzkIbhGQaPHJjmIBzNrKeR9Zc8zeoJGuM+4JlUf2Y4sg6e2JmrEUFB2yA=)
	at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
	at io.kestra.plugin.aws.s3.S3Service.download(S3Service.java:43)
	at io.kestra.plugin.aws.s3.Downloads.lambda$run$0(Downloads.java:96)
	at io.kestra.core.utils.Rethrow.lambda$throwFunction$4(Rethrow.java:90)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
	at io.kestra.plugin.aws.s3.Downloads.run(Downloads.java:100)
	at io.kestra.plugin.aws.s3.Downloads.run(Downloads.java:21)
	at io.kestra.core.runners.Worker$WorkerThread.run(Worker.java:645)
Caused by: software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 400, Request ID: 7M5CBX6N4PHA05PS, Extended Request ID: 2KQkAT2rX5aX4za54NbzkIbhGQaPHJjmIBzNrKeR9Zc8zeoJGuM+4JlUf2Y4sg6e2JmrEUFB2yA=)
```

the documentation on this task could be improved too e.g. "Downloads files to a S3 bucket." should be "Downloads multiple files from an S3 bucket."

### Steps To Reproduce

_No response_

### Environment Information

- Kestra Version: 0.11.0



### Example flow

_No response_

--- a/src/main/java/io/kestra/plugin/aws/s3/Downloads.java
+++ b/src/main/java/io/kestra/plugin/aws/s3/Downloads.java
@@ -38,7 +38,7 @@ import static io.kestra.core.utils.Rethrow.throwFunction;
     }
 )
 @Schema(
-    title = "Downloads files to a S3 bucket."
+    title = "Downloads multiple files from a S3 bucket."
 )
 public class Downloads extends AbstractS3Object implements RunnableTask<List.Output>, ListInterface, ActionInterface {
     private String prefix;
@@ -99,7 +99,7 @@ public class Downloads extends AbstractS3Object implements RunnableTask<List.Out
                 .stream()
                 .map(throwFunction(object -> {
                     GetObjectRequest.Builder builder = GetObjectRequest.builder()
-                        .bucket(bucket)
+                        .bucket(runContext.render(bucket))
                         .key(object.getKey());
 
                     Pair<GetObjectResponse, URI> download = S3Service.download(runContext, client, builder.build());

--- a/src/main/java/io/kestra/plugin/aws/s3/List.java
+++ b/src/main/java/io/kestra/plugin/aws/s3/List.java
@@ -1,5 +1,6 @@
 package io.kestra.plugin.aws.s3;
 
+import com.fasterxml.jackson.annotation.JsonInclude;
 import io.kestra.core.models.annotations.Example;
 import io.kestra.core.models.annotations.Plugin;
 import io.kestra.core.models.executions.metrics.Counter;
@@ -75,6 +76,7 @@ public class List extends AbstractS3Object implements RunnableTask<List.Output>,
     @Builder
     @Getter
     public static class Output implements io.kestra.core.models.tasks.Output {
+        @JsonInclude
         @Schema(
             title = "The list of objects"
         )

--- a/src/test/java/io/kestra/plugin/aws/s3/DownloadsTest.java
+++ b/src/test/java/io/kestra/plugin/aws/s3/DownloadsTest.java
@@ -1,13 +1,20 @@
 package io.kestra.plugin.aws.s3;
 
+import io.kestra.core.runners.RunContextFactory;
+import jakarta.inject.Inject;
 import org.junit.jupiter.api.Test;
 import org.testcontainers.containers.localstack.LocalStackContainer;
+
+import java.util.Map;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.endsWith;
 import static org.hamcrest.Matchers.is;
 
 class DownloadsTest extends AbstractTest {
+    @Inject
+    private RunContextFactory runContextFactory;
+
     @Test
     void delete() throws Exception {
         this.createBucket();
@@ -46,7 +53,7 @@ class DownloadsTest extends AbstractTest {
         Downloads task = Downloads.builder()
             .id(DownloadsTest.class.getSimpleName())
             .type(Downloads.class.getName())
-            .bucket(this.BUCKET)
+            .bucket("{{bucket}}")
             .endpointOverride(localstack.getEndpointOverride(LocalStackContainer.Service.S3).toString())
             .accessKeyId(localstack.getAccessKey())
             .secretKeyId(localstack.getSecretKey())
@@ -58,7 +65,7 @@ class DownloadsTest extends AbstractTest {
             )
             .build();
 
-        List.Output run = task.run(runContext(task));
+        List.Output run = task.run(runContextFactory.of(Map.of("bucket", this.BUCKET)));
 
         assertThat(run.getObjects().size(), is(2));
 

--- a/src/test/java/io/kestra/plugin/aws/s3/ListTest.java
+++ b/src/test/java/io/kestra/plugin/aws/s3/ListTest.java
@@ -1,13 +1,26 @@
 package io.kestra.plugin.aws.s3;
 
+import io.kestra.core.serializers.JacksonMapper;
 import io.kestra.core.utils.IdUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.junit.jupiter.api.Test;
 
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.*;
 
 class ListTest extends AbstractTest {
+    @Test
+    void emptyFolderAlwaysSerialized() throws Exception {
+        this.createBucket();
+
+        List task = list()
+            .build();
+        List.Output run = task.run(runContext(task));
+        assertThat(run.getObjects().size(), is(0));
+
+        assertThat(JacksonMapper.ofJson().writeValueAsString(run), is("{\"objects\":[]}"));
+    }
+
     @Test
     void run() throws Exception {
         this.createBucket();

